### PR TITLE
Documentation on using persistent volumes for pods

### PR DESCRIPTION
**Pull request by meln1k**
Summary: update of the persistent volumes page with additional section regarding its usage in pods.

JIRA issues: MARATHON-7973

### CODE REVIEW

File: docs/docs/persistent-volumes.md [hunk 1]
```diff
@@ -69,6 +69,58 @@ The default `UpgradeStrategy` for a stateful application is a `minimumHealthCapa
...
+    "type": "root",
+    "size": 10,
+    "constraints": []
+  }
+}
+```
+
+- `name`: Name of the pod level volume
+- `persistent.type`: The type of mesos disk resource to use; the valid options are `root`, `path`, and `mount`, corresponding to the [valid mesos multi-disk resource types](http://mesos.apache.org/documentation/latest/multiple-disk/).
+- `persistent.size`: The size of the persistent volume in MiBs.
```
Comment (jeschkies) [inline docs/docs/persistent-volumes.md]: Thanks for being explicit on the type 👍

File: docs/docs/persistent-volumes.md [hunk 2]
```diff
@@ -69,6 +69,58 @@ The default `UpgradeStrategy` for a stateful application is a `minimumHealthCapa
...
 
+## Creating a pod with local persistent volumes
+
+### Configuration options
+
+Let's configure a persistent volume with the following options:
+
+```json
+{
+  "name": "pst",
```
Comment (jeschkies) [inline docs/docs/persistent-volumes.md]: Where should this go? I mean within a pod definition. Should this be under `volumes` in the root?

File: docs/docs/persistent-volumes.md [hunk 3]
```diff
@@ -69,6 +69,58 @@ The default `UpgradeStrategy` for a stateful application is a `minimumHealthCapa
...
+- `name`: Name of the pod level volume
+- `persistent.type`: The type of mesos disk resource to use; the valid options are `root`, `path`, and `mount`, corresponding to the [valid mesos multi-disk resource types](http://mesos.apache.org/documentation/latest/multiple-disk/).
+- `persistent.size`: The size of the persistent volume in MiBs.
+- `persistent.maxSize`: (not seen above) For `root` mesos disk resources, the optional maximum size of an exclusive mount volume to be considered.
+- `persistent.profileName`: (not seen above) The storage volume profile. Only volumes with the specified profile are used to launch an application. It this option is not given, any volume (with or without a profile) will be used for launching.
+- `persistent.constraints`: Constraints restricting where new persistent volumes should be created. Currently, it is only possible to constrain the path of the disk resource by regular expression.
+
+
+Next, we also need to set the `residency` node in order to tell Marathon to setup a stateful pod. 
+Currently, the only valid option for this is:
```
Comment (zen-dog) [inline docs/docs/persistent-volumes.md]: ... These parameters are deprecated and will be removed in the near future.

Comment (ichernetsky) [reply docs/docs/persistent-volumes.md]: It is not as simple as that. They are not deprecated yet, not won't be until there is a clear understanding of how to move forward. Certain code pieces depend upon `residency` presence, BTW. Please refer to Tim's doc and the discussion in Slack.

File: docs/docs/persistent-volumes.md (hunk 2 above)
Comment (ichernetsky) [reply docs/docs/persistent-volumes.md]: Yes.

Comment (ichernetsky) [reply docs/docs/persistent-volumes.md]: @meln1k could you please make it more explicit?

File: docs/docs/persistent-volumes.md [hunk 4]
```diff
@@ -69,6 +69,58 @@ The default `UpgradeStrategy` for a stateful application is a `minimumHealthCapa
...
+    "constraints": []
+  }
+}
+```
+
+- `name`: Name of the pod level volume
+- `persistent.type`: The type of mesos disk resource to use; the valid options are `root`, `path`, and `mount`, corresponding to the [valid mesos multi-disk resource types](http://mesos.apache.org/documentation/latest/multiple-disk/).
+- `persistent.size`: The size of the persistent volume in MiBs.
+- `persistent.maxSize`: (not seen above) For `root` mesos disk resources, the optional maximum size of an exclusive mount volume to be considered.
+- `persistent.profileName`: (not seen above) The storage volume profile. Only volumes with the specified profile are used to launch an application. It this option is not given, any volume (with or without a profile) will be used for launching.
```
Comment (ichernetsky) [inline docs/docs/persistent-volumes.md]: App persistent volumes will support profile names too. Could you please add this to the corresponding section too?

File: docs/docs/persistent-volumes.md [hunk 5]
```diff
@@ -69,6 +69,58 @@ The default `UpgradeStrategy` for a stateful application is a `minimumHealthCapa
...
+- `persistent.constraints`: Constraints restricting where new persistent volumes should be created. Currently, it is only possible to constrain the path of the disk resource by regular expression.
+
+
+Next, we also need to set the `residency` node in order to tell Marathon to setup a stateful pod. 
+Currently, the only valid option for this is:
+
+```
+"residency": {
+  "taskLostBehavior": "WAIT_FOREVER",
+  "relaunchEscalationTimeoutSeconds": 3600
```
Comment (ichernetsky) [inline docs/docs/persistent-volumes.md]: According to the code, in order to use persistent volumes, there are two things required:

1. `unreachableStrategy` must be set to `disabled`,
2. `residency` must be defined (surprisingly, none of its properties is actually used throughout the code. Therefore, let's require the following for the time being:

```
"unreachableStrategy": "disabled",
"residency": {
    "taskLostBehaviour": "WAIT_FOREVER"
}
```

Could you please change it accordingly throughout the doc?

**Comment by ichernetsky** [comment]
One last thing. Since `residency` is not going to be added to pods, could you please remove it from the doc? Thank you.

**Comment by ichernetsky** [comment]
Could you please mention that `unreachableStrategy` must be set to `disabled`, and that it is actually required by Marathon, which will complain if it is set to something different?

**Comment by ichernetsky** [comment]
@sascala could you please review it too? Your feedback is most valuable.

File: docs/docs/persistent-volumes.md [hunk 6]
```diff
@@ -69,6 +64,56 @@ The default `UpgradeStrategy` for a stateful application is a `minimumHealthCapa
...
+- `persistent.size`: The size of the persistent volume in MiBs.
+- `persistent.maxSize`: (not seen above) For `root` mesos disk resources, the optional maximum size of an exclusive mount volume to be considered.
+- `persistent.profileName`: (not seen above) The storage volume profile. Only volumes with the specified profile are used to launch an application. It this option is not given, any volume (with or without a profile) will be used for launching.
+- `persistent.constraints`: Constraints restricting where new persistent volumes should be created. Currently, it is only possible to constrain the path of the disk resource by regular expression.
+
+
+Next, we also need to disable `unreachableStrategy` in order to tell Marathon to setup a stateful pod:
+
+```
+"unreachableStrategy": "disabled",
```
Comment (ichernetsky) [inline docs/docs/persistent-volumes.md]: I promise it is the last one. Could you please do the same for apps too? :)

**Pull request merged**